ddtrace/contrib/asgi/middleware.py

```diff
@@ -70,13 +70,16 @@ def _default_handle_exception_span(exc, span):
 def span_from_scope(scope):
     # type: (Mapping[str, Any]) -> Optional[Span]
     """Retrieve the top-level ASGI span from the scope."""
-    return scope.get("datadog", {}).get("request_span")
+    if "request_spans" in scope.get("datadog", {}):
+        try:
+            return scope.get("datadog", {}).get("request_spans")[0]
+        except ValueError:
+            return {}
```

Comment:
```suggestion
    return scope.get("datadog", {}).get("request_spans", [{}])[0]
```
This is perhaps more compact, but I don't know if it's actually better, in terms of possible errors or even performance, and readability.

Reply:
anecdotally i think your suggestion is faster @P403n1x87! and imo it's slightly more readable

Reply:
well, or you go for the slightly longer way:

```python
datadog = scope.get("datadog")
if not datadog:
    return None

request_spans = datadog.get("request_spans", [])
if not request_spans:
    return None

return request_spans[0]
```

Reply:
Ah I was trying to do this earlier but was making a mistake in the syntax so switched it. I just had `[]` instead of `[{}]`.  I think this is pretty readable and I made sure it returns the correct values if either `"datadog"` or `"request_spans"` is missing.